Makefile

```diff
@@ -331,5 +318,21 @@ graphql-docs:
 	cd docs; build-docs; bundle exec rake graphql:compile_docs
 
 ## Build Meshery GraphQl API specifications
-graphql-build:
+graphql-build: dep-check
 	cd server; cd internal/graphql; go run -mod=mod github.com/99designs/gqlgen generate
+
+#-----------------------------------------------------------------------------
+# Dependencies
+#-----------------------------------------------------------------------------
+.PHONY: dep-check
```

Comment:
Here is a suggestion.

Previously, this `dep-check` will check to see if you have a go version that returns this output `go version go1.19.3 darwin/arm64`

By comparing this to the specified `GOVERSION`, it returned an error stating that the go version is not found and needs to make sure that it is installed and available in the `PATH`.

However after some troubleshooting, I realized that the comparison was actually `1.19` versus `go version go1.19.3 darwin/arm64`. Since it does not match, it will fail.

I have a suggestion, which we can expand on.

```bash
GO_MAJOR_VERSION = $(shell go version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f1)
GO_MINOR_VERSION = $(shell go version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f2)
GO_PATCH_VERSION = $(shell go version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f3)
MINIMUM_SUPPORTED_GO_MAJOR_VERSION = 1
MINIMUM_SUPPORTED_GO_MINOR_VERSION = 19
MINIMUM_SUPPORTED_GO_PATCH_VERSION = 1
GO_VERSION_VALIDATION_ERR_MSG = Go version is not supported, please update to at least $(MINIMUM_SUPPORTED_GO_MAJOR_VERSION).$(MINIMUM_SUPPORTED_GO_MINOR_VERSION).$(MINIMUM_SUPPORTED_GO_PATCH_VERSION)

dep-check:
	@if [ $(GO_MAJOR_VERSION) -gt $(MINIMUM_SUPPORTED_GO_MAJOR_VERSION) ]; then \
		exit 0 ;\
	elif [ $(GO_MAJOR_VERSION) -lt $(MINIMUM_SUPPORTED_GO_MAJOR_VERSION) ]; then \
		echo '$(GO_VERSION_VALIDATION_ERR_MSG)';\
		exit 1; \
	elif [ $(GO_MINOR_VERSION) -lt $(MINIMUM_SUPPORTED_GO_MINOR_VERSION) ] ; then \
		echo '$(GO_VERSION_VALIDATION_ERR_MSG)';\
		exit 1; \
	elif [ $(GO_PATCH_VERSION) -lt $(MINIMUM_SUPPORTED_GO_PATCH_VERSION) ] ; then \
		echo '$(GO_VERSION_VALIDATION_ERR_MSG)';\
		exit 1; \
	fi
```

By listing out the version supported, we can ensure that each time someone runs a go command from the `Makefile`, it will run if the version installed on the machine matches the minimum Go version. 

If it doesn't, it should fail and ask to install at least the minimum. The only caveat is that if the installed version is Go 1.18, the script will continue, but the good thing is that if it is at `1.17` and it will fail because it's looking for that specified `go1.17.x` version.

Reply:
> However after some troubleshooting, I realized that the comparison was actually 1.19 versus go version go1.19.3 darwin/arm64. Since it does not match, it will fail.

This comparison is intentional. We're only trying to find a match for `1.19`. Was the `dep-check` for `1.19` (as written in this PR) failing on your 1.19.3 system?

Reply:
And this is even when checking against the substring match for `$(GOVERSION)` set to `1.19` versus `INSTALLED_GO_VERSION` which will return something like `go version 1.19.x arch`

Even when I changed the `GOVERSION` back to `1.19.x`, it still failed.

Reply:
I'm confused. In what situation does a check work for you?